packages/babel-traverse/src/path/modification.js

```diff
@@ -112,7 +116,10 @@ export function insertAfter(nodes) {
     this.parentPath.isExpressionStatement() ||
     this.parentPath.isLabeledStatement()
   ) {
-    return this.parentPath.insertAfter(nodes);
+    // `replaceWithMultiple` requeues if there's a replacement for this.node,
+    // but not for an ancestor's. Set `shouldRequeue` to true so that any replacement
+    // for an ancestor node can be enqueued. Fix #5628 and #5023.
+    return this.parentPath._insertAfter(nodes, true);
```

Comment:
That example will run to [branch 3](http://astexplorer.net/#/gist/47c24345d741602dc04928b353fdd254/081331b15feba9ea524d25707173b478ff5636f2) (i.e. `_containerInsertBefore()`). Still, if we replicate the logic, it will fix the case code goes to branch 1 and then 4, but how cases that go that branch 4 directly?